Documentation/Devices-conventions.md

```diff
@@ -0,0 +1,25 @@
+# Conventions for devices APIs
+
+## Value returned by sensor
+
+- prefer property (i.e. Temperature) rather than method (i.e. GetTemperature)
+- use `double` when you need to return any floating point value
+- use `Vector2/3/4` for returning vectors`*`
+- value should conform to units conventions (see below)
+- only the most useful APIs should be public, anything else which may be useful but will unlikely get used by most of the people should be protected (inheriting the class allows you to use it but it is not visible by default) or internal/private
```

Comment:
I think we should tell more when protected should be used. like API is believed not going to be used much but there is some specific non main stream scenario can be used so making it protected can allow such scenarios to be achieved.

Reply:
I'll let someone else fix the wording and merge as is for now (with casing fix)